### PR TITLE
Android fastfile template: clean before build

### PR DESCRIPTION
**Pull request by slorber**
This ensures that newcomers don't encounter this issue: https://github.com/fastlane/fastlane/issues/4675

### CODE REVIEW

**Comment by slorber** [comment]
I signed it!

**Comment by janpio** [comment]
@slorber PR looks good.

Just to make sure this is actually the correct solution: Why is #4675 actually happening? Is `supply` just getting whatever is in the `build` folder? Or _why_ is the `clean` necessary?

**Comment by slorber** [comment]
@janpio as far as I understand I built my app with Android Studio which generated a debug apk and the default Fastlane android template did try assemble release (so there are now debug + release apk) and upload debug + release.

In my case it failed because the versionCode of the debug apk was already existing, because it was not incremented.

I think it's decent to cleanup build folder before assembling new release.

Other solution could be to make sure the release build is uploaded.

Don't really know if it makes sense to upload a debug apk to google play store, I'm a newbie Android developer (React Native actually)

**Comment by janpio** [comment]
Ok, this might point to a deeper problem with `supply` (now called `upload_to_play_store`) in the way it picks up APKs. Will look into this (or at least try to understand what it does).

(I am not qualified to decide if a `clean` before each build is useful. Hope someone else will look into this.)

**Comment by ohayon** [comment]
Oh, hai, @fastlane/android, any chance you folks could take a peek at this? 🤖

File: fastlane/lib/assets/FastfileTemplateAndroid [hunk 1]
```diff
@@ -27,6 +27,7 @@ platform :android do
 
   desc "Submit a new Beta Build to Crashlytics Beta"
   lane :beta do
+    gradle(task: 'clean')
     gradle(task: "assembleRelease")
```
Comment (casz) [inline fastlane/lib/assets/FastfileTemplateAndroid]: You could make into one Gradle action `gradle(task: "clean assembleRelease")`

**Comment by slorber** [comment]
@casz thanks I've updated the pr

**Comment by janpio** [comment]
Hi @slorber, sorry that you had to wait so long for a reaction. 

Unfortunately CI is failing with a failing test, maybe something caused by the additional clean? Also there is a merge conflict now that has to be taken care of. Could you please rebase onto `master`? (This will also retrigger a CI run with current CI setup which might give more information.)

**Comment by slorber** [comment]
@janpio I've created a new PR https://github.com/fastlane/fastlane/pull/11873 because the code has been refactored since last PR.

**Pull request closed without merging**